rotkehlchen/api/rest.py

```diff
@@ -3029,18 +3028,19 @@ def _get_ethereum_transactions(
             self,
             only_cache: bool,
             filter_query: ETHTransactionsFilterQuery,
+            event_params: Dict[str, Any],
     ) -> Dict[str, Any]:
         tx_module = EthTransactions(
             ethereum=self.rotkehlchen.chain_manager.ethereum,
             database=self.rotkehlchen.data.db,
         )
-        transactions: Optional[List[EthereumTransaction]]
         try:
             transactions, total_filter_count = tx_module.query(
                 only_cache=only_cache,
                 filter_query=filter_query,
                 has_premium=self.rotkehlchen.premium is not None,
             )
+            transactions = list(set(transactions))
```

Comment:
Why are you doing this here?

Reply:
Hm... Looks like I've forgotten about `DISTINCT` keyword in SQL :))